tools/fraken/main.go

```diff
@@ -0,0 +1,451 @@
+			case strings.HasPrefix(trimmedT, "extension = "):
+				tokens := strings.Split(t, " ")
+				extension = strings.Replace(tokens[len(tokens)-1], `"`, "", -1)
+			case strings.HasPrefix(trimmedT, "owner = "):
+				tokens := strings.Split(t, " ")
+				owner = strings.Replace(tokens[len(tokens)-1], `"`, "", -1)
+			}
+		}
+		if trimmedT == "meta:" {
+			if meta {
+				log.Fatal("error funging Yara rule: was in meta section and met meta section header")
+			}
+			meta = true
+		}
+		if condition {
+			if filepath != "" {
+				if filepath[0] == '!' {
+					not = "not "
+					filepath = filepath[1:]
+				}
+				t = t + fmt.Sprintf(" and %vfilepath matches /%v/", not, filepath)
+				filepath = ""
+				not = ""
+			}
+			if filename != "" {
+				if filename[0] == '!' {
+					not = "not "
+					filename = filename[1:]
+				}
+				t = t + fmt.Sprintf(" and %vfilename matches /%v/", not, filename)
+				filename = ""
+				not = ""
+			}
+			if filetype != "" {
+				if filetype[0] == '!' {
+					not = "not "
+					filetype = filetype[1:]
+				}
+				t = t + fmt.Sprintf(` and %vfiletype == "%v"`, not, filetype)
+				filetype = ""
+				not = ""
+			}
+			if extension != "" {
+				if extension[0] == '!' {
+					not = "not "
+					extension = extension[1:]
+				}
+				if !strings.HasPrefix(extension, ".") {
+					extension = "." + extension
+				}
+				t = t + fmt.Sprintf(` and %vextension == "%v"`, not, extension)
+				extension = ""
+				not = ""
+			}
+			if owner != "" {
+				if owner[0] == '!' {
+					not = "not "
+					owner = owner[1:]
+				}
+				t = t + fmt.Sprintf(` and %vowner == "%v"`, not, owner)
+				owner = ""
+				not = ""
+			}
+			condition = false
+		}
+		if trimmedT == "condition:" {
+			if condition {
+				log.Fatal("error funging Yara rule: was in condition section and met condition header")
+			}
+			condition = true
+		}
+		ret = append(ret, t)
+	}
+	return strings.Join(ret, "\n")
+}
+
+// Compile will compile the provided Yara rules in a Rules object.
+func (s *Scanner) compile() error {
+	compiler, err := yara.NewCompiler()
+	if err != nil {
+		return err
+	}
+	for _, v := range externalVariables {
+		compiler.DefineVariable(v, "")
+	}
+
+	rulesStat, _ := os.Stat(s.rulesPath)
+	switch mode := rulesStat.Mode(); {
+	case mode.IsDir():
+		err = filepath.Walk(s.rulesPath, func(filePath string, fileInfo os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			fileName := fileInfo.Name()
+
+			// Check if the file has extension .yar or .yara.
+			if (path.Ext(fileName) == ".yar") || (path.Ext(fileName) == ".yara") {
+				// log.Println("Adding rule ", filePath)
+
+				// Open the rule file and add it to the Yara compiler.
+				err = compiler.AddString(fungeRules(filePath), "")
+				if err != nil {
+					log.Fatalf("unable to parse rule %v: %v", filePath, err)
+				}
+			}
+			return nil
+		})
+		if err != nil {
+			log.Printf("error walking the path %v\n", err)
```

Comment:
Since you're already returning error from this function it would be better to do

return fmt.Errorf("error walking the path %v, err)

Reply:
Done